MapboxCoreNavigation/CLLocation.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import CoreLocation
+import MapboxDirections
+
+public extension CLLocation {
+    /**
+     Returns a `CLLocation` infront of the provided location given the interval
+     with the current speed taken into account.
+     */
+    public func advance(by interval: TimeInterval) -> CLLocation {
+        let metersAhead = speed * interval
+        let locationInfrontOfUser = coordinate.coordinate(at: metersAhead, facing: course)
+        return CLLocation(coordinate: locationInfrontOfUser.coordinate(at: metersAhead, facing: course),
+                          altitude: altitude,
+                          horizontalAccuracy: horizontalAccuracy,
+                          verticalAccuracy: verticalAccuracy,
+                          course: course,
+                          speed: speed,
+                          timestamp: timestamp.addingTimeInterval(interval))
+    }
+    
+    public func isOnRoute(_ routeProgress: RouteProgress) -> Bool {
+        let newLocation = advance(by: RouteControllerDeadReckoningTimeInterval)
+        let radius = CLLocation.rerouteRadius(horizontalAccuracy)
+        let isCloseToCurrentStep = newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
+        
+        // If the user is moving away from the maneuver location
+        // and they are close to the next step
+        // we can safely say they have completed the maneuver.
+        // This is intended to be a fallback case when we do find
+        // that the users course matches the exit bearing.
+        if let upComingStep = routeProgress.currentLegProgress.upComingStep {
+            let isCloseToUpComingStep = newLocation.isWithin(radius, of: upComingStep)
+            if !isCloseToCurrentStep && isCloseToUpComingStep {
+                return !isCloseToCurrentStep && isCloseToUpComingStep
+            }
+        }
+        
+        return isCloseToCurrentStep
+    }
+    
+    /**
+     Returns the calculated reroute distance.
+     */
+    public static func rerouteRadius(_ horizontalAccuracy: CLLocationAccuracy) -> CLLocationDistance {
```

Comment:
Replace `_` with `with` or `for`. `_` would only be appropriate if `horizontalAccuracy` could be called a “radius”.

Also, “reroute” is an imperative form, suggesting that this method actually reroutes something. Call this method `radiusForRerouting(with:)`.